attr/validators.py

```diff
@@ -85,3 +85,33 @@ def provides(interface):
     the value it got.
     """
     return _ProvidesValidator(interface)
+
+
+@attributes(repr=False)
+class _OptionalValidator(object):
+    """
+    A validator that makes an attribute optional.  An optional field is one
+    which can be set to None in addition to its intended data type.
+    """
+    v = attr()
+
+    def __call__(self, inst, attr, value):
+        if value is None:
+            return
+        return self.v(inst, attr, value)
+
+    def __repr__(self):
+        return (
+            "<optional validator for {type} or None>"
+            .format(type=repr(self.v))
+        )
+
+
+def optional(x):
```

Comment:
I’m not enthusiastic about `x` as the parameter name.  Let's call it `validator` which it is. :)